asio/include/xtt/asio/server_context.inl

```diff
@@ -0,0 +1,334 @@
+/******************************************************************************
+ *
+ * Copyright 2018 Xaptum, Inc.
+ * 
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ * 
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License
+ *
+ *****************************************************************************/
+
+#include <xtt/asio/error_category.hpp>
+
+namespace xtt {
+namespace asio {
+
+    template <typename GPKLookupCallback,
+              typename AssignIdCallback,
+              typename Handler>
+    void
+    server_context::async_do_read(std::tuple<GPKLookupCallback, AssignIdCallback, Handler> func_pack)
+    {
+        boost::asio::async_read(socket_,
```

Comment:
Not now, but we should find a new indenting style that doesn't spill so far past the 100 column mark.

Reply:
Ha, yea, all the ASIO callback style stuff really frustrates me; I like the GNU/C-style "no longer than 80 characters" lines...

Definitely agree.

Reply:
Much of the ASIO example code just indents one "tab" (4 spaces, ...) on the next line.  I need to find the emacs `c-mode` style that works like that, but we could switch to that for these callback-heavy functions.